torchinductor/lowering.py

```diff
@@ -631,14 +631,15 @@ def rand_or_randn(
             size = [sympy.expand(s) for s in size]
             offset = V.graph.increment_randomness_offset(sympy_product(size))
 
+            random_pos = ir.FixedLayout(
+                device,
+                dtype,
+                size,
+                ir.FlexibleLayout.contiguous_strides(size),
+                offset=offset,
+            ).make_indexer()
+
             if device.type == "cuda":
```

Comment:
Can we remove this condition entirely?

Reply:
Good question! I am actually ignorant of the logic here. Let me try. What I don't know is - can we use the same seed for both? ATM, the cuda seed is from `ops.load(V.graph.random_seed_buffer(device), sympy.Integer(0))` and the cpu seed is from `ops.index_expr(V.graph.sizevars.seed(), torch.int32)`. Ostensibly, yes, since `random_seed_buffer` looks like it should just work with a dtype. 

Let me try it.